musit-models/src/main/scala/no/uio/musit/models/Module.scala

```diff
@@ -0,0 +1,25 @@
+package no.uio.musit.models
+
+import play.api.libs.json.{JsNumber, Reads, Writes, __}
+
+sealed trait Module {
+  val id: Int
+}
+
+case object StorageFacility extends Module {
+  val id = 1
+}
+case object LoanProcess extends Module {
+  val id = 2
+}
+
+object Module {
+
+  implicit val reads: Reads[Module]   = __.read[Int].map(fromInt)
+  implicit val writes: Writes[Module] = Writes{m => JsNumber(m.id)}
+
+  def fromInt(id: Int): Module = id match {
+    case StorageFacility.id => StorageFacility
```

Comment:
Database mapper is only one point in the code where `fromInt` is used. There's also the implicit reads for Module.
We cannot 100% control the quality of data being sent into the API. And handling the parsing correctly is essential for correctness and being able to provide a sensible error message to the client.

The `@throws` annotation is for clarity that the `unsafeFromInt` function is just that, unsafe. It is the `get` operation on the `Option` that throw the actual Exception. If you want to be explicit about what kind of exception you want to throw, an `IllegalArgumentException` would be OK.

```
implicit val reads: Reads[Module] = Reads { jsv =>
  jsv.validate[Int] match {
    case JsSuccess(id, path) =>
      fromInt(id).map(JsSuccess.apply).getOrElse(JsError(path, s"Unknown Module ID $id"))

    case err: JsError =>
      err
  }
}
```

So please change as initially described. (Including the `reads` from this comment)